datman/xnat.py

```diff
@@ -88,260 +89,399 @@ class xnat(object):
         """
 
-        url = "{server}/data/archive/projects/{project}/" \
-              "subjects/{subject}/experiments/{experiment}" \
-              "?xsiType=xnat:mrSessionData" \
-              .format(server=self.server,
-                      project=project,
-                      subject=session,
-                      experiment=experiment)
+        url = "{}/data/archive/projects/{}/subjects/{}/experiments/" \
+              "{}?xsiType=xnat:mrSessionData".format(
+                  self.server, project, subject, experiment)
+        try:
+            self._make_xnat_put(url)
+        except requests.exceptions.RequestException as e:
+            raise XnatException("Failed to create XNAT experiment {} under "
+                                "subject {} in project {}. Reason - {}".format(
+                                    experiment, subject, project, e))
 
-        self._make_xnat_put(url)
+    def get_scan_ids(self, project, subject, experiment):
+        """Retrieve all scan IDs for an XNAT experiment.
+
+        Args:
+            project (:obj:`str`): An XNAT project ID.
+            subject (:obj:`str`): An existing subject within 'project'.
+            experiment (:obj:`str`): An existing experiment within 'subject'.
+
+        Raises:
+            XnatException: If server/API access fails.
+
+        Returns:
+            list: A list of scan IDs belonging to 'experiment'.
+        """
+        logger.debug("Querying XNAT server {} for scan IDs belonging to "
+                     "experiment {} of subject {} in project {}".format(
+                         self.server, experiment, subject, project))
+
+        url = "{}/data/archive/projects/{}/subjects/{}/experiments/{}" \
+              "/scans/?format=json".format(
+                  self.server, project, subject, experiment)
 
-    def get_scan_list(self, study, session, experiment):
-        """The list of dicom scans in an experiment"""
-        url = '{}/data/archive/projects/{}' \
-              '/subjects/{}/experiments/{}' \
-              '/scans/?format=json'.format(self.server,
-                                           study,
-                                           session,
-                                           experiment)
         try:
             result = self._make_xnat_query(url)
         except Exception:
-            raise XnatException('Failed getting scans with url: {}'
-                                ''.format(url))
+            raise XnatException("Failed getting scan IDs for experiment {} "
+                                "with URL {}".format(experiment, url))
 
-        if result is None:
-            e = XnatException('Scan not found for experiment: {}'
-                              .format(experiment))
-            e.study = study
-            e.session = session
-            raise e
+        if not result:
+            return []
+
+        return [item.get("ID") for item in result["ResultSet"]["Result"]]
```

Comment:
```suggestion
        return [item["ID"] for item in result["ResultSet"]["Result"]]
```
same as earlier suggestion